### DISCUSSION TITLE
Telemetry

### DISCUSSION THREAD

**Discussion by ghost** [App-vNext/Polly#1740] [Q&A]
Hi,

I want to add telemetry and logging to Polly. I have read this:

https://www.pollydocs.org/advanced/telemetry.html

but I dont get how this works:

    LoggerFactory = LoggerFactory.Create(builder => builder.AddConsole())

Have someone tried this and can expalin?

Bjorn

**Comment by martincostello** [comment]
The `LoggerFactory.Create(builder => builder.AddConsole())` part creates an [`ILoggerFactory`](https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-library-authors#when-to-use-the-iloggerfactory-interface) instance which is configured to be able to log to the console.

The `ILoggerFactory` is then passed to the `TelemetryOptions` used to configure telemetry within Polly as part of the `ConfigureTelemetry()` call to the `ResiliencePipelineBuilder`.

At runtime, Polly will then use the `ILoggerFactory` to create `ILogger` instances which are used internally to emit log messages [such as those shown here](https://www.pollydocs.org/advanced/telemetry.html#logs).

**Reply by ghost** [reply]
Thanks for the very fast response!
